lib/puppetfactory/plugins/gitea.rb

```diff
@@ -18,16 +18,19 @@ def initialize(options)
     @admin_password = options[:gitea_admin_password] || 'puppetlabs'
     @gitea_port     = options[:gitea_port]           || '3000'
     @gitea_user     = options[:gitea_user]           || 'git'
+    @gitea_homedir  = Dir.home(@gitea_user)
 
-    migrate_repo! unless File.directory?(@repopath)
+    migrate_repo! unless File.directory?(@cache_dir)
```

Comment:
@binford2k The previous version checked to see if the end user's repository existed inside the cache directory, but this change checks for the parent cache dir itself. Won't that prevent `migrate_repo!` from being called after the first repository is migrated?

Reply:
There's only a single repository. This is only used to create the user branches.

Reply:
Got it, I knew that but brain cramped :)

But shouldn't `@repopath` be checked instead of `@cache_dir` anyway? Maybe not a problem if the cache dir can never be present without the full path to the repository also present? Or am I misreading it?

Reply:
It's to minimize the race condition. The `@cache_dir` is created immediately when `migrate_repo!` is called, but `@repopath` is only created once git does some indeterminate stuff and creates it. That's also why I moved the `mkdir_p` call to the beginning of the method.